### PR TITLE
Add endpoint to allow for a manual settlement for a given period

### PR DESCRIPTION
**Pull request by Timo614**
### References Issues:
Relates to https://app.asana.com/0/1138868228851596/1132788783646657

### Description:
It was mentioned in mattermost it would be useful to trigger manual settlements. We've been able to do it by hand but this allows the team to test settlements in a much more conductive way. I didn't attach a frontend component here but it should be easy for team members to trigger these via the API as well.

```
-- Get auth header token value:
curl --header "Content-Type: application/json" --request POST --data '{"email":"superuser@test.com","password":"password"}' http://localhost:9000/auth/token -i

-- Create settlement report
curl --header "Content-Type: application/json" --request POST --data '{"start_time": "2019-10-21T18:00:00", "end_time": "2019-10-21T22:00:00"}' http://localhost:9000/organizations/99dd51f8-cf48-4193-904f-c1d953090923/settlements -i --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxYzZkNGIyZS1kNTE5LTRiOGEtYTgzZS04ZmYxODVhZDI0YjEiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTcxNjgzMDgwfQ.AHoOGgn68eLAmRMYam4YykvQygoDhbsKnSgWDYpVi98"
HTTP/1.1 201 Created
content-length: 326
content-type: application/json
access-control-allow-origin: http://localhost:3000
access-control-expose-headers: x-app-version
vary: Origin
x-app-version: 1.0.272
date: Mon, 21 Oct 2019 18:32:45 GMT

{"id":"a00ece3b-c1aa-451b-ac74-686d27a7039d","organization_id":"99dd51f8-cf48-4193-904f-c1d953090923","start_time":"2019-10-21T18:00:00","end_time":"2019-10-21T22:00:00","status":"PendingSettlement","comment":null,"only_finished_events":true,"created_at":"2019-10-21T18:32:45.875688","updated_at":"2019-10-21T18:32:45.875688"}
```

### Migrations
 * No Migrations

### Environment Variables
 * No change

### CODE REVIEW

File: api/src/controllers/settlements.rs [hunk 1]
```diff
@@ -27,6 +36,29 @@ pub fn index(
...
+    user.requires_scope_for_organization(Scopes::SettlementWrite, &organization, connection)?;
+    let new_settlement = Settlement::create(
+        organization.id,
+        new_settlement.start_time,
+        new_settlement.end_time,
+        SettlementStatus::PendingSettlement,
+        new_settlement.comment.clone(),
+        organization.settlement_type == SettlementTypes::PostEvent,
+    );
+    let settlement = new_settlement.commit(connection)?;
```
Comment (mikethetike) [inline api/src/controllers/settlements.rs]: Maybe add the user in here to pass to any domain events that are created. 
```suggestion
    let settlement = new_settlement.commit(user.user, connection)?;
```

Comment (Timo614) [reply api/src/controllers/settlements.rs]: Sounds good, will do.

Comment (Timo614) [reply api/src/controllers/settlements.rs]: All set,

```
select * from domain_events order by created_at desc limit 1;
                  id                  |        event_type         |     display_text     |                        event_data                        |  main_table   |               main_id                |         created_at         |         updated_at         |               user_id                | seq  | organization_id 
--------------------------------------+---------------------------+----------------------+----------------------------------------------------------+---------------+--------------------------------------+----------------------------+----------------------------+--------------------------------------+------+-----------------
 80fb9655-1ea9-42ab-b28f-c756d4c84e33 | SettlementReportProcessed | Settlement processed | {"settlement_id":"3232e6c7-bdd9-47cf-985f-939daff39046"} | Organizations | cf1fea01-8a9a-4218-8a47-7ef4670217cd | 2019-10-22 16:16:15.149872 | 2019-10-22 16:16:15.149872 | bff9c54a-fa3e-40e9-b15b-610a1a0c7d1d | 2899 | 
(1 row)
```

User is now passed in for the manual settlements so the associated domain event contains their ID.

**Comment by Timo614** [comment]
Just pushed up a fix for my fmt issue -- had updated to rust 1.38.0 when I noticed build issues this morning since I had a slightly older version but didn't realize we locked our rust version for the drone down to 1.30.0 so had some fmt differences. Switched my local machine to 1.33.0 and it's properly fmting now.

**Pull request merged**